command/agentproxyshared/cache/cachememdb/index.go

```diff
@@ -76,12 +77,43 @@ type Index struct {
 	// LastRenewed is the timestamp of last renewal
 	LastRenewed time.Time
 
-	// Type is the index type (token, auth-lease, secret-lease)
+	// Type is the index type (token, auth-lease, secret-lease, static-secret)
 	Type string
 
+	// Capabilities is a set of known capabilities for the given token. Used only for
+	// token-capabilities type cache entries.
+	// Implemented as a map for uniqueness.
+	Capabilities map[string]struct{}
+
 	// IndexLock is a lock held for some indexes to prevent data
 	// races upon update.
-	IndexLock sync.Mutex
+	IndexLock sync.RWMutex
+}
+
+// CapabilitiesIndex holds the capabilities for cached static secrets.
+// This type of index does not represent a response.
+type CapabilitiesIndex struct {
+	// ID is a value that uniquely represents the request held by this
+	// index. This is computed by hashing the token that this capabilities
+	// index .
+	// Required: true, Unique: true
+	ID string
+
+	// Token is the token that fetched the response held by this index
+	// Required: true, Unique: true
+	Token string
+
+	// Type is the index type (token-capabilities)
+	Type string
+
+	// Capabilities is a set of known capabilities for the given token. Used only for
+	// token-capabilities type cache entries.
+	// Implemented as a map for uniqueness.
+	Capabilities map[string]struct{}
```

Comment:
This might be more space efficient as a radix tree, if we're at all worried about a single token fetching many secrets that may have a lot of common prefix, e.g. many secrets from the same kv mount.